uniparc-common/src/main/java/org/uniprot/api/uniparc/common/repository/store/crossref/UniParcCrossReferenceFacetConfig.java

```diff
@@ -0,0 +1,238 @@
+                facets.add(status);
+            }
+        }
+
+        if (facetNames.contains(UniParcCrossReferenceFacet.ORGANISMS.getFacetName())) {
+            Facet organisms =
+                    Facet.builder()
+                            .label(UniParcCrossReferenceFacet.ORGANISMS.getLabel())
+                            .name(TAXONIDS_FIELD)
+                            .allowMultipleSelection(false)
+                            .values(getFacetItemOrganisms(lightCrossReferences))
+                            .build();
+            if (Utils.notNullNotEmpty(organisms.getValues())) {
+                facets.add(organisms);
+            }
+        }
+
+        if (facetNames.contains(UniParcCrossReferenceFacet.DATABASES.getFacetName())) {
+            Facet databases =
+                    Facet.builder()
+                            .label(UniParcCrossReferenceFacet.DATABASES.getLabel())
+                            .name(DBTYPES_FIELD)
+                            .allowMultipleSelection(false)
+                            .values(getFacetItemDatabases(lightCrossReferences))
+                            .build();
+            if (Utils.notNullNotEmpty(databases.getValues())) {
+                facets.add(databases);
+            }
+        }
+
+        return facets;
+    }
+
+    private UniParcCrossReference getLightCrossReference(UniParcCrossReference crossReference) {
+        UniParcCrossReferenceBuilder builder = new UniParcCrossReferenceBuilder();
+        builder.active(crossReference.isActive());
+        builder.database(crossReference.getDatabase());
+        builder.organism(crossReference.getOrganism());
+        return builder.build();
+    }
+
+    List<FacetItem> getFacetItemStatuses(List<UniParcCrossReference> crossReferences) {
+        Map<String, Long> statusCount =
+                crossReferences.stream()
+                        .collect(
+                                Collectors.groupingBy(
+                                        crossRef -> crossRef.isActive() ? ACTIVE_STR : INACTIVE_STR,
+                                        Collectors.counting()));
+        Map<String, String> labelNameMap =
+                Map.of(ACTIVE_STR, Boolean.TRUE.toString(), INACTIVE_STR, Boolean.FALSE.toString());
+        return statusCount.entrySet().stream()
+                .sorted(Map.Entry.<String, Long>comparingByValue().reversed())
+                .map(entry -> getStatusFacetItem(entry, labelNameMap))
+                .toList();
+    }
+
+    List<FacetItem> getFacetItemOrganisms(List<UniParcCrossReference> crossReferences) {
+        // taxonId to count map
+        Map<Long, Long> taxonIdCount =
+                crossReferences.stream()
+                        .filter(crossRef -> Objects.nonNull(crossRef.getOrganism()))
+                        .collect(
+                                Collectors.groupingBy(
+                                        crossRef -> crossRef.getOrganism().getTaxonId(),
```

Comment:
Do you think here that we could return an Organism object and this way we would not need taxonIdName map?